src/plugins/footnotes/widgets.ts

```diff
@@ -139,8 +137,36 @@ export const deleteFootnoteWidget =
             })
           }
         }
+        // Apply the transaction to delete the footnote
 
         view.dispatch(tr)
+
+        // Check if all footnotes within `footnotesElement` have been deleted and remove the footnotesElement if true
+
+        if (pos) {
+          const footnotesElement = findParentNodeClosestToPos(
+            view.state.doc.resolve(pos),
+            (node) => node.type === schema.nodes.footnotes_element
+          )
+
+          if (footnotesElement) {
+            let allFootnotesDeleted = true
+
+            // Check if all footnote in footnotesElement are deleted
+            footnotesElement.node.forEach((child) => {
+              if (child.type === schema.nodes.footnote && !isDeleted(child)) {
+                allFootnotesDeleted = false
+              }
+            })
+            // If all footnotes are deleted, remove the entire footnotes element
+            if (allFootnotesDeleted) {
+              const { pos: fnPos, node: fnNode } = footnotesElement
+              const tr2 = view.state.tr // Create a new transaction to delete the footnotesElement
+              tr2.delete(fnPos, fnPos + fnNode.nodeSize + 1)
+              view.dispatch(tr2) // Dispatch the transaction to delete footnotesElement
+            }
```

Comment:
why not dispatch only once at the end of the handleDelete func.

Reply:
@fwaisi 
The logic for deleting the footnotesElement`based on whether all footnotes are deleted won't functioning because the check occurs before the current transaction (tr) for deletion is applied. This means the verification for remaining footnotes happens prior to the actual deletion. To resolve this, we need to apply the transaction to delete the specific footnote first, and then check if all footnotes in the footnotesElement have been deleted. This approach ensures that the document state is updated, allowing the check to accurately reflect the current status of all footnotes.

Reply:
but If you used the current tr variable you will be able to access the latest state of the doc which will have the deleted footnote, would you please check.

Reply:
I’ve already tried it, but it’s not efficient or effective. You need to click the delete button again or refresh the page to see the changes accurately reflected.